lib/lte_link_control/lte_lc_helpers.c

```diff
@@ -990,6 +993,16 @@ int parse_ncellmeas(const char *at_response, struct lte_lc_cells_info *cells)
 		goto clean_exit;
 	}
 
+	__ASSERT_NO_MSG(cells->neighbor_cells != NULL);
```

Comment:
This seems unnecessary considering line 992.

Reply:
Yes and no. The caller allocates `cells->neighbor_cells` structure based on `neighborcell_count_get()` return value and we use the same function here for `cells->ncells_count`. I think asserts are good for double checking in these conditions that everything is alright when the values of two variables are related to each other, but you may well do a mistake during development.

Reply:
I was looking at diff for the added commit, not the whole PR. In the diff there was a `NULL` check for `cells->neighbor_cells` before the assert, which seems to be removed in the following commit. So, my comment can be ignored.